lwt/client/dns_client_lwt.ml

```diff
@@ -1,58 +1,69 @@
-(* {!Transport} provides the implementation of the underlying flow
-   that is in turn used by {!Dns_client.Make} to provide the
-   Lwt convenience module
+(* {!Transport} provides an implementation of Dns_client.S used by
+   {!Dns_client.Make} to provide a DNS client which uses Lwt
+   TODO clarify, linse and hannes understand this comment
 *)
 
 open Lwt.Infix
 
 module Transport : Dns_client.S
-  with type flow = Lwt_unix.file_descr
- and type io_addr = Lwt_unix.inet_addr * int
+ with type io_addr = Lwt_unix.inet_addr * int
  and type +'a io = 'a Lwt.t
  and type stack = unit
 = struct
   type io_addr = Lwt_unix.inet_addr * int
-  type flow = Lwt_unix.file_descr
   type ns_addr = [`TCP | `UDP] * io_addr
   type +'a io = 'a Lwt.t
   type stack = unit
   type t = {
-    rng : (int -> Cstruct.t) ;
     nameserver : ns_addr ;
+    timeout_ns : int64 ;
   }
+  type context = { t : t ; fd : Lwt_unix.file_descr ; timeout_ns : int64 ref }
 
   let create
-    ?(rng = Dns_client.stdlib_random)
-    ?(nameserver = `TCP, (Unix.inet_addr_of_string Dns_client.default_resolver, 53)) () =
-    { rng ; nameserver }
+    ?(nameserver = `TCP, (Unix.inet_addr_of_string Dns_client.default_resolver, 53))
+    ~timeout () =
+    Mirage_crypto_rng_unix.initialize ();
```

Comment:
This silently overwrites the global `Mirage_crypto_rng.generator` ref, which is pretty unintuitive and feels a bit drastic (and is not documented in the mli). I think it would be preferable to document that, or to only do this if it is currently uninitialized?

Reply:
This is side-effecty but since it is a sensible default we are ok with that. And we are gonna document it in the `.mli` like you proposed.